Contents/Code/youtube_dl_helper.py

```diff
@@ -41,74 +44,97 @@ def process_youtube(url):
-        youtube_include_dash_manifest=False,
-    )
-
-    ydl = youtube_dl.YoutubeDL(params=youtube_dl_params)
-
-    with ydl:
-        try:
-            result = ydl.extract_info(
-                url=url,
-                download=False  # We just want to extract the info
-            )
-        except Exception as exc:
-            if isinstance(exc, youtube_dl.utils.ExtractorError) and exc.expected:
-                Log.Info('YDL returned YT error while downloading %s: %s' % (url, exc))
-            else:
-                Log.Exception('YDL returned an unexpected error while downloading %s: %s' % (url, exc))
-            return None
-
-        if 'entries' in result:
-            # Can be a playlist or a list of videos
-            video_data = result['entries'][0]
-        else:
-            # Just a video
-            video_data = result
-
-    selected = {
-        'opus': {
-            'size': 0,
-            'audio_url': None
-        },
-        'mp4a': {
-            'size': 0,
-            'audio_url': None
-        },
-    }
-    if video_data:
-        for fmt in video_data['formats']:  # loop through formats, select largest audio size for better quality
-            if 'audio only' in fmt['format']:
-                if 'opus' == fmt['acodec']:
-                    temp_codec = 'opus'
-                elif 'mp4a' == fmt['acodec'].split('.')[0]:
-                    temp_codec = 'mp4a'
+
+    with tempfile.NamedTemporaryFile() as cookie_jar_file:
+        youtube_dl_params = dict(
+            cookiefile=cookie_jar_file.name,
+            logger=plugin_logger,
+            outtmpl=u'%(id)s.%(ext)s',
+            password=Prefs['str_youtube_passwd'] if Prefs['str_youtube_passwd'] else None,
+            socket_timeout=10,
+            username=Prefs['str_youtube_user'] if Prefs['str_youtube_user'] else None,
+            youtube_include_dash_manifest=False,
+        )
+
+        if Prefs['str_youtube_cookies']:
+            try:
+                cookies = json.loads(Prefs['str_youtube_cookies'])
+                cookie_jar_file.write('# Netscape HTTP Cookie File\n')
+                for cookie in cookies:
+                    include_subdom = True if cookie['domain'].startswith('.') else False
+                    expiry = int(cookie.get('expiry', 0))
+                    nsbool = lambda b: 'TRUE' if b else 'FALSE'
+                    values = [
+                        cookie['domain'],
+                        nsbool(include_subdom),
+                        cookie['path'],
+                        nsbool(cookie['secure']),
+                        str(expiry),
+                        cookie['name'],
+                        cookie['value']
+                    ]
+                    cookie_jar_file.write('\t'.join(values) + '\n')
+            except Exception as exc:
+                Log.Exception('Failed to write YouTube cookies to file, will try anyway. Error: %s' % exc)
```

Comment:
```suggestion
            except Exception as e:
                Log.Exception('Failed to write YouTube cookies to file, will try anyway. Error: {}'.format(e))
```

I'm slowly working towards replacing all `%s` with `.format()` when possible.